### DISCUSSION TITLE
0.3.1 compilation warnings on linux

### DISCUSSION THREAD

**Discussion by marlowa** [jeremy-rifkin/cpptrace#68] [General]
I think that a recent change for MVC has introduced compiler warnings when that compiler is not in use, e.g. on linux.
Here is the output I see:

/home/marlowa/mystuff/cpptrace-stuff/cpptrace-0.3.1/src/symbols/../binary/../utils/utils.hpp:368:13: warning: "_MSC_VER" is not defined, evaluates to 0 [-Wundef]
  368 |         #if _MSC_VER != 1938
      |             ^~~~~~~~
/home/marlowa/mystuff/cpptrace-stuff/cpptrace-0.3.1/src/symbols/../binary/../utils/utils.hpp:418:13: warning: "_MSC_VER" is not defined, evaluates to 0 [-Wundef]
  418 |         #if _MSC_VER != 1938
      |             ^~~~~~~~
[ 89%] Building CXX object CMakeFiles/cpptrace-lib.dir/src/symbols/symbols_with_nothing.cpp.o
[ 90%] Building CXX object CMakeFiles/cpptrace-lib.dir/src/symbols/symbols_core.cpp.o
In file included from /home/marlowa/mystuff/cpptrace-stuff/cpptrace-0.3.1/src/symbols/../binary/object.hpp:5,
                 from /home/marlowa/mystuff/cpptrace-stuff/cpptrace-0.3.1/src/symbols/symbols.hpp:9,
                 from /home/marlowa/mystuff/cpptrace-stuff/cpptrace-0.3.1/src/symbols/symbols_core.cpp:1:
/home/marlowa/mystuff/cpptrace-stuff/cpptrace-0.3.1/src/symbols/../binary/../utils/utils.hpp:368:13: warning: "_MSC_VER" is not defined, evaluates to 0 [-Wundef]
  368 |         #if _MSC_VER != 1938
      |             ^~~~~~~~
/home/marlowa/mystuff/cpptrace-stuff/cpptrace-0.3.1/src/symbols/../binary/../utils/utils.hpp:418:13: warning: "_MSC_VER" is not defined, evaluates to 0 [-Wundef]
  418 |         #if _MSC_VER != 1938
      |             ^~~~~~~~

**Comment by jeremy-rifkin** [comment]
Hi @marlowa! Thanks for posting this, it will be fixed in the next release! The commit just after the 0.3.1 tag fixes this, 56b50d279aa84e55fb5bdd6b783ad812bc14b46f, if you'd like to get the fix sooner :)

**Reply by marlowa** [reply]
Thank you. Just a suggestion, if the build treated warnings as errors then I think it would reduce the chance of this sort of thing slipping through.

**Reply by jeremy-rifkin** [reply]
That's a good idea, I'll see about turning that on for the CI pipelines at the very least
